### DISCUSSION TITLE
Hard fork https://github.com/JetBrains/spotbugs-intellij-plugin

### DISCUSSION THREAD

**Discussion by hazendaz** [spotbugs/spotbugs#2773] [General]
Its very clear we are getting no where with intellij.  I think we should hard-fork this and try our best to get it up to date and figure out the release process under the spotbugs umbrella.  No one over there seems to respond.  They have some 900+ tickets mostly for same issue to which they introduced themselves in intellij.  I think it would be of our best interest to just hard fork it, figure it out, when people raise issues, try to get them to help, etc.

What do others think here? I must say I know nothing about intellij really as I use Eclipse and I'd know much less abut plugins but I have had to patch their error prone plugin for corporate usage as it sucks too by insistent need to access the web.  So maybe this is not much more than that other than figuring out publishing.

**Comment by ThrawnCA** [comment]
It's been a long time since I used IntelliJ at all, I mostly work in Geany. The fact that complex IDEs require a whole extra project to be able to integrate with tools like this, instead of just running the standard command-line build, has a lot to do with why...

**Comment by iloveeclipse** [comment]
I must confess I've never used intellij and do not plan to do in the near future :-)
I barely have time to look into Eclipse issues (but I'm there for sure fix any blockers), however I would likely have no time/interest to help here.

**Comment by JuditKnoll** [comment]
As far as I can see, @gtoison tried to reach [@jgyp](https://github.com/jqyp) at https://github.com/spotbugs/spotbugs/issues/2763#issuecomment-1854611248, who is the creator of the [SpotBugs IntelliJ Plugin](https://github.com/JetBrains/spotbugs-intellij-plugin) (according to the readme), and @hazendaz tried to reach the code owners at https://github.com/JetBrains/spotbugs-intellij-plugin/issues/1572.
Both without success so far, but it was only a few days ago, and it's the second half of December, which can be quite busy. It looks like, @jqyp was not so active in the last half year on GitHub, so I'm not hopeful about receiving an answer from him.

All the releases were created by @amaembo, who still seems to be active in GitHub, just not in the [SpotBugs IntelliJ Plugin](https://github.com/JetBrains/spotbugs-intellij-plugin).
@amaembo Can you please share some info about the project? Do you or any of its codeowners plan to continue its development? If not, can you please help us with the release and some other processes?

The release seems less automated than it is in SpotBugs core, I couldn't find any documentation or CI in connection with it at first glance, but I didn't make a thorough search.
What may be a challange is the release of the plugin to the IntelliJ IDEA plugin marketplace and making users to transition to that. 
Currently there is only two hits in the IntelliJ IDEA plugin marketplace for "SpotBugs": 
- SpotBugs IntelliJ Plugin itself: https://plugins.jetbrains.com/plugin/14014-spotbugs
- Reshift Security, which depends on the former: https://plugins.jetbrains.com/plugin/14538-reshift-security

I currently use IntelliJ for development, but I have no experience of developing IntelliJ plugins. However, I'm not sure I would have time for the project.
IMO if we take over the project, there would be much more work with it, there are several issues with the project itself, and even finding the relevant issues from the pile of auto-generated ones might take a lot of time and effort. Also, we have plenty of issues for SpotBugs core itself.
I think before doing anything about the project, we could wait at least a month so everyone have time to react and get back from the holiday season.

**Comment by gtoison** [comment]
I have created an issue here: https://youtrack.jetbrains.com/issue/IDEA-341351/The-SpotBugs-plugin-seems-abandonned
Hopefully this will get more visibility than posting issues on the project issue tracker

My understanding is that the plugin is now broken and won't run at all with the latest version of IntelliJ: https://github.com/JetBrains/spotbugs-intellij-plugin/issues/1556

**Comment by gtoison** [comment]
Seems like the plugin got some love again, I think we can close this discussion

**Reply by gtoison** [reply]
Looks like I spoke too fast: https://youtrack.jetbrains.com/issue/IDEA-341351#focus=Comments-27-8764718.0-0

> Note that it would still be better to fork the plugin. I cannot promise that I will support it in the future.

**Reply by hazendaz** [reply]
Yes sorry meant to come back here and comment back.  I'm not sure we need to fork it at this point.  Clearly though based on the comment its not likely to get attention again.  I know nothing about how to release a plugin like that.  It would need some work to become ours and we would want them to at least make their copy read only down the road.  It must have been your ticket that drove getting most attention.  I heard back from them on mine but I'm sure only because they were closing 100s of tickets :)

That said, what do we want to do, do we want to simply fork it for now and do nothing?  If things break again then we move?

**Reply by gtoison** [reply]
The release part is easy: you build a zip file an upload it to their marketplace. Then Jetbrains does a review and after a few days the new version is visible to the users.
However taking the time to setup and understand yet another project would be the limiting factors, unless new contributors step in.

On the bright side, the IntelliJ plugin API is really well done and documented!
